### PR TITLE
Changed Default Member Privilege to Pending

### PR DESCRIPTION
**Pull request by ThaiQ**
Changing the default access-level to pending (-1)

### CODE REVIEW

**Comment by ThaiQ** [comment]
What member see when they go to profile
image

**Comment by ThaiQ** [comment]
Travis failed because default user is no longer admin - hence no access to any/most api-routing-functions

**Comment by ThaiQ** [comment]
"Not Valid" is not red
image

**Comment by ThaiQ** [comment]
do you have any example of mocking login?

**Comment by evanugarte** [comment]
### What I did

Added a mocking class for test cases. Using `APIFunctions` is helpful as not only do we abstract axios, we can also mock the output from those requests.

The class derives itself from [this existing example](https://github.com/SCE-Development/Core-v4/blob/d3a10ddf59949ed92d147dbf303a5d08c3ff3c08/test/frontend/EventList.test.js#L30)

### How to Fix the PR
1. Open `3DPrinting.js` in `test/`
2. Open `3DPrinting.js` in `api/routes/`
3. Using console logs and [cool google searches like this](https://lmgtfy.com/?q=mocking+a+function+sinon&pp=1) find out how to make the token verify function return true for when we need it to

File: api/routes/Event.js [hunk 1]
```diff
@@ -27,10 +27,9 @@ router.post('/createEvent', (req, res) => {
   } else if (!checkIfTokenValid(req)) {
     return res.sendStatus(UNAUTHORIZED)
   }
-  logger.log(req.body.id)
   const newEvent = new Event({
     title: req.body.title,
-    description: req.body.description,
+    description: req.body.dcheckIfTokenValidescription,
```
Comment (evanugarte) [inline api/routes/Event.js]: can we keep this as `description`?

File: api/routes/Event.js [hunk 2]
```diff
@@ -27,10 +27,9 @@ router.post('/createEvent', (req, res) => {
   } else if (!checkIfTokenValid(req)) {
     return res.sendStatus(UNAUTHORIZED)
   }
-  logger.log(req.body.id)
   const newEvent = new Event({
     title: req.body.title,
```
Comment (evanugarte) [inline api/routes/Event.js]: fun fact, by doing

```js
const obj = {
   title
}
```

this automatically implies
```js
const obj = {
   title: title
}
```

change this `NewEvent` line by doing
```js
const {
  title, eventLocation, eventDate,
  /* all other fields here */
} = req.body
```
then do

```js
 const newEvent = new Event({
  title, eventLocation, eventDatem
   /* all other fields here */
})
```

File: src/Pages/Profile/MemberView/InfoCard.js [hunk 3]
```diff
@@ -53,7 +53,8 @@ export default function ProfileCard (props) {
       <div id='profile-box'>
         {props.fields.map((field, ind) => (
           <h3 key={ind} id='inner-text-top'>
-            <b>{field.title}:</b> {field.value}
+            <b>{field.title}:</b>{' '}
+            <nobr dangerouslySetInnerHTML={{ __html: field.value }} />
```
Comment (evanugarte) [inline src/Pages/Profile/MemberView/InfoCard.js]: why are you doing this line? i dont like injecting values into HTML. it leads to [xss](https://www.youtube.com/watch?v=L5l9lSnNMxg)

File: src/Pages/Profile/MemberView/Profile.jsx [hunk 4]
```diff
@@ -40,7 +40,9 @@ export default class Profile extends Component {
           { title: 'Door Code', value: this.state.user.doorCode },
           { title: 'Joined Date', value: this.state.user.joinDate.slice(0, 10) },
           { title: 'Email', value: this.state.user.email },
-          { title: 'Membership Expiration', value: this.state.user.membershipValidUntil.slice(0, 10) },
+          { title: 'Membership Expiration', value: (this.props.user.accessLevel<0)? 
+          "<font color='red'>Not Valid</font>, please see us at Engr294" :
```
Comment (evanugarte) [inline src/Pages/Profile/MemberView/Profile.jsx]: make the whole line red instead
```suggestion
          "<font color='red'>Not Valid, please see us at Engr294</font>" :
```

File: test/mocks/TokenValidFunctions.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+function restoreMock () {
+  checkifTokenValidMock.restore()
+}
+
+function resetMock () {
+  checkifTokenValidMock.reset()
+}
+
+function setReturnOfTokenValidMock (returnValue) {
```
Comment (evanugarte) [inline test/mocks/TokenValidFunctions.js]: could you name this something that reads more like english? e.g.

```suggestion
function tokenValidWillReturn (returnValue) {
```

File: test/mocks/TokenValidFunctions.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+const TokenFunctions = require('../../util/api-utils/token-functions')
+const sinon = require('sinon')
+
+const checkifTokenValidMock = sinon.stub(TokenFunctions, 'checkIfTokenValid')
+
+function restoreMock () {
```
Comment (evanugarte) [inline test/mocks/TokenValidFunctions.js]: add a simple comment to each function in this file, for this one you can do 

```
/**
 * This function unstubs the checkIfTokenValid function and is called at the end of each
 * test.
 */
```

File: src/Pages/Profile/MemberView/InfoCard.js (hunk 3 above)
Comment (evanugarte) [reply src/Pages/Profile/MemberView/InfoCard.js]:
```html
<b>{field.title}:</b>{' '}
<span className={field.something ? 'red-text' : ''}>{field.value}</span>
```

File: test/mocks/TokenValidFunctions.js (hunk 5 above)
Comment (evanugarte) [reply test/mocks/TokenValidFunctions.js]: consider doing this^

Comment (ThaiQ) [reply test/mocks/TokenValidFunctions.js]: done

File: api/models/User.js [hunk 7]
```diff
@@ -59,7 +59,7 @@ const UserSchema = new Schema(
     // 2: Admin
     accessLevel: {
       type: Number,
-      default: 2
+      default: -1
```
Comment (evanugarte) [inline api/models/User.js]: no more hardcoded numbers. [we have enums for this](https://github.com/SCE-Development/Core-v4/blob/3c4ba959ba6a462c28bae5841594edf1ed625033/src/Enums.js#L28) so import them

File: api/routes/user.js [hunk 8]
```diff
@@ -107,7 +107,7 @@ router.post('/login', function (req, res) {
           if (isMatch && !error) {
             if (
               new Date() - user.membershipValidUntil > 0 &&
-              user.accessLevel < 2
+              user.accessLevel < 0
```
Comment (evanugarte) [inline api/routes/user.js]: no more hardcoded numbers. [we have enums for this](https://github.com/SCE-Development/Core-v4/blob/3c4ba959ba6a462c28bae5841594edf1ed625033/src/Enums.js#L28) so import them

File: src/Pages/Profile/MemberView/Profile.jsx [hunk 9]
```diff
@@ -40,7 +40,9 @@ export default class Profile extends Component {
           { title: 'Door Code', value: this.state.user.doorCode },
           { title: 'Joined Date', value: this.state.user.joinDate.slice(0, 10) },
           { title: 'Email', value: this.state.user.email },
-          { title: 'Membership Expiration', value: this.state.user.membershipValidUntil.slice(0, 10) },
+          { title: 'Membership Expiration', value: (this.props.user.accessLevel<0)?
```
Comment (evanugarte) [inline src/Pages/Profile/MemberView/Profile.jsx]: no more hardcoded numbers. [we have enums for this](https://github.com/SCE-Development/Core-v4/blob/3c4ba959ba6a462c28bae5841594edf1ed625033/src/Enums.js#L28) so import them

File: api/routes/user.js [hunk 10]
```diff
@@ -108,7 +108,7 @@ router.post('/login', function (req, res) {
           if (isMatch && !error) {
             if (
               new Date() - user.membershipValidUntil > 0 &&
-              user.accessLevel < membershipState.ADMIN
+              user.accessLevel < membershipState.PENDING
```
Comment (evanugarte) [inline api/routes/user.js]: nice. but can you explicitly say that this applies to banned people with
```suggestion
              user.accessLevel === membershipState.BANNED
```

File: src/Pages/Profile/MemberView/Profile.jsx (hunk 9 above)
Comment (evanugarte) [reply src/Pages/Profile/MemberView/Profile.jsx]: BRUH

Comment (ThaiQ) [reply src/Pages/Profile/MemberView/Profile.jsx]: forgot to commit

**Pull request merged**